test/surface_test.py

```diff
@@ -815,41 +814,81 @@ def test_blit__SRCALPHA_opaque_source(self):
         for pt in test_utils.rect_area_pts(src.get_rect()):
             self.assertEqual(dst.get_at(pt)[1], src.get_at(pt)[1])
 
-    def todo_test_blit__blit_to_self(self):  # TODO
-        src = pygame.Surface((256, 256), SRCALPHA, 32)
-        rect = src.get_rect()
+    def test_blit__blit_to_self(self):
+        """Test that blit operation works on self, and that no distortion occurs. """
+        test_surface = pygame.Surface((128, 128), SRCALPHA, 32)
+        area = test_surface.get_rect()
+
+        for pt, test_color in test_utils.gradient(area.width, area.height):
+            test_surface.set_at(pt, test_color)
 
-        for pt, color in test_utils.gradient(rect.width, rect.height):
-            src.set_at(pt, color)
+        reference_surface = test_surface.copy()
 
-        src.blit(src, (0, 0))
+        test_surface.blit(test_surface, (0, 0))
 
-    def todo_test_blit__SRCALPHA_to_SRCALPHA_non_zero(self):  # TODO
-        # " There is no unit test for blitting a SRCALPHA source with non-zero
-        #   alpha to a SRCALPHA destination with non-zero alpha " LL
+        # The below does not work, need to resolve. #TODO
```

Comment:
@MyreMylar This fails here. Not sure why.

Reply:
Just tested it manually and it's failing because of alpha compositing (blending a surface with non 255 alpha to another surface with non 255 alpha). 

The test should fail because you wouldn't expect a surface with alpha pixels blitted on top of another with the same to produce the same surface with the default alpha belnd mode (or most alpha blend modes). Perhaps this test should use an opaque gradient instead?

This is the utility function in test_utils:

```
def gradient(width, height):
    """

    Yields a pt and corresponding RGBA tuple, for every (width, height) combo.
    Useful for generating gradients.
    
    Actual gradient may be changed, no tests rely on specific values.
    
    Used in transform.rotate lossless tests to generate a fixture.

    """

    for l in xrange_(width):
        for t in xrange_(height):
            yield (l, t), tuple(map(rgba_between, (l, t, l, l + t)))
```

the bit that says `l + t` would have to be changed to 255 in a new `opaque_gradient()` utility function.

Reply:
I changed the alpha value expression to more general form. Thanks for the heads up on the non-opaque gradient.